src/models/scatter.js

```diff
@@ -448,20 +451,22 @@ nv.models.scatter = function() {
                     .type(function(d) { return getShape(d[0]); })
                     .size(function(d) { return z(getSize(d[0],d[1])) })
             );
-            points.exit().remove();
+            points.exit().each(delCache).remove();
             groups.exit().selectAll('path.nv-point')
                 .watchTransition(renderWatch, 'scatter exit')
                 .attr('transform', function(d) {
                     return 'translate(' + nv.utils.NaNtoZero(x(getX(d[0],d[1]))) + ',' + nv.utils.NaNtoZero(y(getY(d[0],d[1]))) + ')'
                 })
                 .remove();
-            points.filter(function (d) { return scaleDiff || sizeDiff || getDiffs(d, 'x', 'y'); })
+            // Update points position only if "x" or "y" have changed
+            points.filter(function (d) { return scaleDiff || sizeDiff || getDiffs(d, 'x', getX, 'y', getY); })
```

Comment:
It would be nice and more atomic if I could just pass the getter functions on their own, but there needs to be a way to identify the function uniquely to store it's value in the cache for comparison next time.

The getDiffs function is generic and takes n key + getter pairs in any order, the key can actually be anything, so long as the same key is used for referring to the same value in any subsequent calls.

I did look at the options for using functions as keys: ES6 has a nice reliable function.name property which refers to the _first_ reference assigned to the function regardless of scope and var vs function name... ES6 and under is unreliable without just turning the whole function source into a string which seems a bit excessive... Sorry long explanation.